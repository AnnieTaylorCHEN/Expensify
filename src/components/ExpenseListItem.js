import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'


export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h2>{description}</h2>
        </Link>
        <p>Amount: {amount}</p>
        <p>Created at: {createdAt}</p>  
    </div>
)

export default connect()(ExpenseListItem)