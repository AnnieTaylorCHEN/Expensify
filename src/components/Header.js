import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from "react-redux"
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => (
    <div>
        <header>
            <h1>Expensify</h1>
        </header>
        <NavLink to="/" activeClassName="is-active" exact >Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        <button onClick={startLogout} >Logout</button>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)