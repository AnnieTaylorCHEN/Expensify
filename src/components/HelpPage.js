import React from 'react'

const HelpPage = () => (
    <div>
    this is HelpExpensePage
    </div>
)


export default HelpPage