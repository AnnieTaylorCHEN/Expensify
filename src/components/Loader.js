import React from 'react'

const Loader = () => (
    <div className="loader">
        <img className="loader__img" src="/images/loader.svg" />
    </div>
)

export default Loader