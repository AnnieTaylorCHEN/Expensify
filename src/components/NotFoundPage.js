import React from 'react'
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
    <div>
    404 page not found. Did you type something wrong? <Link to="/">Go back Home.</Link>
    </div>
)

export default NotFoundPage