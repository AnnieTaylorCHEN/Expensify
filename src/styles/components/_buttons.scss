.button {
    background: $purple;
    color: white;
    font-weight: 300;
    line-height: 1;
    text-decoration: none;
    padding: $s-size $m-size;
    border: none;
    display: inline-block; 
}

.button--danger {
    background: red;
}