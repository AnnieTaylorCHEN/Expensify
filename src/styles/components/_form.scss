.form {
    display: flex;
    flex-direction: column;
     > * {
         margin-bottom: $m-size;
     }
}

.form__error {
    color: red;
    margin: 0 0 $m-size 0;
    font-style: italic;
}
