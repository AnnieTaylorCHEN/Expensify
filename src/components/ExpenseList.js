import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import SelectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--msg">
                        <span>No Expenses</span>
                    </div>
                    
                ) : (props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />
                    })
                )
            }  
        </div>     
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: SelectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList)

