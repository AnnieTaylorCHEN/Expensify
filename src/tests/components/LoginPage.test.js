import React from 'react'
import {shallow} from 'enzyme'
import { LoginPage } from '../../components/LoginPage'

test('should correctly render login page', () => {
    const wrapper = shallow(<LoginPage />)
    expect(wrapper).toMatchSnapshot()
})

test('should call log in after button clicked', () => {
    const LoginSpy = jest.fn()
    const wrapper = shallow(<LoginPage startLogin={LoginSpy}/>)
    wrapper.find('button').simulate('click')
    expect(LoginSpy).toHaveBeenCalled() 
})