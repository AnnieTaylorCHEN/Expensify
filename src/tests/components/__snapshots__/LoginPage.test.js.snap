// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render login page 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
